### PR TITLE
fix: should not have have read access to all secrets when secret targets disabled

### PR DESCRIPTION
**Pull request by erikgb**
After https://github.com/cert-manager/trust-manager/pull/193 trust-manager needs read RBAC to secrets cluster-wide, even if secret targets support is disabled.

### CODE REVIEW

File: cmd/trust-manager/app/options/options.go [hunk 1]
```diff
@@ -152,6 +152,10 @@ func (o *Options) addBundleFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&o.Bundle.DefaultPackageLocation,
 		"default-package-location", "",
 		"Path to a JSON file containing the default certificate package. If set, must be a valid package.")
+
+	fs.BoolVar(&o.Bundle.SecretTargetsEnabled,
+		"secret-targets-enabled", false,
+		"Path to a JSON file containing the default certificate package. If set, must be a valid package.")
```
Comment (SgtCoDFish) [inline cmd/trust-manager/app/options/options.go:158]: note: copy paste needs updating for the description here

Comment (erikgb) [reply cmd/trust-manager/app/options/options.go]: Thanks! 😊 Fixed now.

File: deploy/charts/trust-manager/templates/clusterrole.yaml [hunk 2]
```diff
@@ -44,24 +44,24 @@ rules:
...
-{{- if .Values.secretTargets.authorizedSecretsAll }}
+  verbs: ["get", "list", "create", "patch", "watch", "delete"]
+{{- else if .Values.secretTargets.authorizedSecrets }}
 - apiGroups:
   - ""
   resources:
   - "secrets"
-  verbs: ["get", "update", "delete", "patch"]
-{{- else if .Values.secretTargets.authorizedSecrets }}
+  verbs: ["list", "watch"]
```
Comment (erikgb) [inline deploy/charts/trust-manager/templates/clusterrole.yaml:59]: I know this is probably least-privilege permissions, but it also makes the template more complex. I don't see a major benefit of granting `list`, but not `get`, cluster-wide. The controller needs RBAC to read all secrets cluster-wide if secret-targets is enabled - even if it doesn't use the `get` verb.

Comment (inteon) [reply deploy/charts/trust-manager/templates/clusterrole.yaml:59]: I moved "get" to the other list.

File: deploy/charts/trust-manager/templates/clusterrole.yaml [hunk 3]
```diff
@@ -44,24 +44,24 @@ rules:
...
-{{- if .Values.secretTargets.authorizedSecretsAll }}
+  verbs: ["get", "list", "create", "patch", "watch", "delete"]
+{{- else if .Values.secretTargets.authorizedSecrets }}
 - apiGroups:
   - ""
   resources:
   - "secrets"
-  verbs: ["get", "update", "delete", "patch"]
-{{- else if .Values.secretTargets.authorizedSecrets }}
+  verbs: ["list", "watch", "get"]
```
Comment (erikgb) [inline deploy/charts/trust-manager/templates/clusterrole.yaml:59]:
```suggestion
  verbs: ["get", "list", "watch"]
```

Nit: For consistency

**Review by erikgb — COMMENTED** [review]
Thanks for picking this up @inteon! Almost LGTM from me. Just a minor nit comment.

**Review by erikgb — COMMENTED** [review]
/lgtm

(I cannot approve my own PR)

Maybe rebase/squash before merge?

**Pull request merged**